src/components/PatchActionButtons/EnqueuePatch.tsx

```diff
@@ -23,6 +32,19 @@ export const EnqueuePatch: React.VFC<EnqueuePatchProps> = ({
       ? visibilityControl
       : fallbackVisibilityControl;
 
+  const { data, previousData } = useQuery<
+    CodeChangesQuery,
+    CodeChangesQueryVariables
+  >(GET_CODE_CHANGES, {
+    variables: { id: patchId },
+    skip: !isVisible,
+  });
+  const { patch } = data ?? previousData ?? {};
```

Comment:
I did this to prevent this glitchy weirdness 😞

https://user-images.githubusercontent.com/47064971/167157497-c1d21f1d-b9b9-4f93-94e5-808a26ac2d9b.mov

Reply:
How does this prevent the modal from doing that?

Reply:
When the state of the modal is no longer visible the `data` gets reset to `undefined` so `preserveCommits` becomes interpreted as `false`

Something interesting is that if I keep the `antd` Modal (as opposed to this LG Modal), the glitchiness doesn't happen

Reply:
I experimented by replacing our `VersionRestartModal` (which has a similar `skip: !visible` query) with a LG Modal as well and the same thing happens 😞

https://user-images.githubusercontent.com/47064971/167202303-61f62823-b5cf-4d6f-939a-0cb28cb20369.mov

Reply:
interesting... i would be curious to know why this happens. Does Apollo set the data to undefined when the component unmounts? Does LG unmount the modal in a different way than  antd?

Reply:
Yeah 🤔 Maybe `antd` preserves the content somehow before the modal fully closes? They do have a swanky animation on their modal which possibly presents the user from noticing it